packages/firestore/src/platform_browser/browser_platform.ts

```diff
@@ -34,10 +35,14 @@ export class BrowserPlatform implements Platform {
   }
 
   get document(): Document | null {
+    // `document` is not always available, e.g. in ReactNative and WebWorkers.
+    // eslint-disable-next-line no-restricted-globals
     return typeof document !== 'undefined' ? document : null;
   }
 
   get window(): Window | null {
+    // `window` is not always available, e.g. in ReactNative and WebWorkers.
```

Comment:
Drive-by comment: should this fall back on `self` in a web worker?

Reply:
I don't know and we should be careful if we want to make this change. FWIW, this doc says to always use `self`: https://developer.mozilla.org/en-US/docs/Web/API/Window/self

This works because JS in browsers prefixes any unknown globals with `window` and `window.self` seems to be a thing.